lib/src/st2110/st_tx_video_session.c

```diff
@@ -423,10 +425,25 @@ static int tv_train_pacing(struct mtl_main_impl* impl, struct st_tx_video_sessio
     reactive = (s->ops.height == 480) ? 487.0 / 525.0 : 576.0 / 625.0;
   }
   pkts_per_frame = pkts_per_frame * reactive;
-  if (pkts_per_frame < s->st20_total_pkts) {
-    err("%s(%d), error pkts_per_frame %f, st20_total_pkts %d\n", __func__, idx,
-        pkts_per_frame, s->st20_total_pkts);
-    return -EINVAL;
+  measured_bps = s->st20_pkt_size * pkts_per_sec * reactive;
+
+  /* If the measured speed is lower than expected. Set higher bps and retrain to add
+   * padding */
+  if (measured_bps < rl_bps) {
+    info("%s(%d), measured bps %" PRIu64 " is lower then set bps %" PRIu64 "\n", __func__,
+         idx, (uint64_t)measured_bps, rl_bps);
+
+    if (!mt_pacing_train_bps_result_search(impl, port, rl_bps, &bps_to_set)) {
+      err("%s(%d), measured speed is too low on already trained bps\n", __func__, idx);
+      return -EINVAL;
+    }
+#define INCREASE_BPS_FACTOR 1.005
```

Comment:
please add a comment why INCREASE_BPS_FACTOR  is needed.

Reply:
Fixed in 87490591608abf90b86cf59a793c5c47e13952ca